haystack/schema.py

```diff
@@ -629,6 +637,11 @@ def to_dict(self):
 
     @classmethod
     def from_dict(cls, dict: dict):
+        # backward compatibility for old labels using answers with document_id instead of document_ids
+        answer = dict.get("answer")
+        if answer and "document_id" in answer:
+            dict = dict.copy()
+            dict["answer"] = Answer.from_dict(dict["answer"])
```

Comment:
I added a test for this in `test_schema.py` and for all search_engine and sql document stores.